src/lib/int.ts

```diff
@@ -498,47 +643,82 @@ class Int64 extends CircuitValue implements BalanceChange {
     return xInt;
   }
 
+  /**
+   * Negates the value.
+   *
+   * `Int64.from(5).neg()` will turn into `Int64.from(-5)`
+   */
   neg() {
     // doesn't need further check if `this` is valid
     return new Int64(this.magnitude, this.sgn.neg());
   }
-
+  /**
+   * Addition with overflow checking.
+   */
   add(y: Int64 | number | string | bigint | UInt64 | UInt32) {
     let y_ = Int64.from(y);
     return Int64.fromField(this.toField().add(y_.toField()));
   }
+  /**
+   * Subtraction with underflow checking.
+   */
   sub(y: Int64 | number | string | bigint | UInt64 | UInt32) {
     let y_ = Int64.from(y);
     return Int64.fromField(this.toField().sub(y_.toField()));
   }
+  /**
+   * Multiplication with overflow checking.
+   */
   mul(y: Int64 | number | string | bigint | UInt64 | UInt32) {
     let y_ = Int64.from(y);
     return Int64.fromField(this.toField().mul(y_.toField()));
   }
+  /**
+   * Integer division.
+   *
+   * `x.div(y)` returns the floor of `x / y`, that is, the greatest
+   * `z` such that `x * y <= x`.
```

Comment:
this comment and the original one on UInt{32,64} has a typo, should be
"the greatest `z` such that `z * y <= x`"